### ISSUE TITLE
Analysis fails because field of TypeInfoImpl has type incompatible with proxy of XmlSeeAlso

### ISSUE THREAD

**Issue by argenstijn** [quarkusio/quarkus#14652]
**Describe the bug**
While building a native image using quarkus the analysis stage fails with the reason:

    Error: Field com.sun.xml.bind.v2.model.impl.TypeInfoImpl.upstream has declared type com.sun.xml.bind.v2.model.annotation.Locatable which is incompatible with types in state: 1TypeMObject<[0x00005C0F00005C10:!S:XmlSeeAlso$$ProxyImpl]>

In my project, I can only find com.sun.xml.bind.v2.model.impl.TypeInfoImpl in the jar org.glassfish.jaxb:jaxb-runtime:jar:2.3.3-b02:compile in my dependencies, which is only pulled in within the following top-level dependency:

 +- io.quarkus:quarkus-hibernate-orm-panache:jar:1.6.1.Final:compile
 |  +- io.quarkus:quarkus-hibernate-orm:jar:1.6.1.Final:compile
 |  |  +- org.hibernate:hibernate-core:jar:5.4.18.Final:compile
 |  |  |  +- net.bytebuddy:byte-buddy:jar:1.10.10:compile
 |  |  |  +- antlr:antlr:jar:2.7.7:compile
 |  |  |  +- org.dom4j:dom4j:jar:2.1.3:compile
 |  |  |  \- org.hibernate.common:hibernate-commons-annotations:jar:5.1.0.Final:compile
 |  |  +- org.hibernate:hibernate-graalvm:jar:5.4.18.Final:compile
 |  |  +- org.glassfish.jaxb:jaxb-runtime:jar:2.3.3-b02:compile
 |  |  |  +- org.glassfish.jaxb:txw2:jar:2.3.3-b02:compile
 |  |  |  \- com.sun.istack:istack-commons-runtime:jar:3.0.10:compile
 |  |  +- org.jboss.spec.javax.xml.bind:jboss-jaxb-api_2.3_spec:jar:2.0.0.Final:compile
 |  |  +- jakarta.persistence:jakarta.persistence-api:jar:2.2.3:compile
 |  |  +- org.hibernate:quarkus-local-cache:jar:0.1.0:compile
 |  |  |  \- com.github.ben-manes.caffeine:caffeine:jar:2.8.4:compile
 |  |  \- io.quarkus:quarkus-caffeine:jar:1.6.1.Final:compile
 |  +- io.quarkus:quarkus-hibernate-orm-panache-common:jar:1.6.1.Final:compile
 |  \- io.quarkus:quarkus-panache-common:jar:1.6.1.Final:compile

[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Running Quarkus native-image plugin on GraalVM Version 20.1.0 (Java Version 11.0.7)                                                                                                                    
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] /home/myhome/tools/graalvm/bin/native-image -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-DCoordinatorEnvironmentBean.transactionStatusManagerEnable=false -J-Djava.util.logging.manager=org.jboss.logmanage
r.LogManager -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=1 -J-Duser.language=en -J-Dfile.encoding=
UTF-8 -H:DynamicProxyConfigurationFiles=proxies.json -H:DynamicProxyConfigurationFiles=proxies_resources.json -H:+TraceClassInitialization --initialize-at-build-time= -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy$BySpaceAndTime -H:
+JNI -jar identity-1.0.0-SNAPSHOT-runner.jar -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:-AddAllCharsets -H:-IncludeAllTimeZones -H:EnableURLProtocols=http,https --enable-all-security-services -H:NativeLinkerOption=-no-pie --no-server -H:-UseServic
eLoaderFeature -H:+StackTrace identity-1.0.0-SNAPSHOT-runner                                                                                                                                                                                                         
-H:IncludeAllTimeZones and -H:IncludeTimeZones are now deprecated. Native-image includes all timezonesby default.                                                                                                                                                    
[identity-1.0.0-SNAPSHOT-runner:134700]    classlist:  23,905.79 ms,  1.62 GB                                                                                                                                                                                        
[identity-1.0.0-SNAPSHOT-runner:134700]        (cap):   1,279.26 ms,  2.54 GB                                                                                                                                                                                        
[identity-1.0.0-SNAPSHOT-runner:134700]        setup:   3,521.45 ms,  2.54 GB                                                                                                                                                                                        
17:10:01,379 INFO  [org.hib.val.int.uti.Version] HV000001: Hibernate Validator 6.1.5.Final                                                                                                                                                                           
17:10:01,597 INFO  [org.hib.Version] HHH000412: Hibernate ORM core version 5.4.18.Final                                                                                                                                                                              
17:10:01,604 INFO  [org.hib.ann.com.Version] HCANN000001: Hibernate Commons Annotations {5.1.0.Final}                                                                                                                                                                
17:10:01,647 INFO  [org.hib.dia.Dialect] HHH000400: Using dialect: io.quarkus.hibernate.orm.runtime.dialect.QuarkusPostgreSQL10Dialect                                                                                                                               
17:10:58,398 INFO  [org.jbo.threads] JBoss Threads version 3.1.1.Final                                                                                                                                                                                               
WARNING GR-10238: VarHandle for static field is currently not fully supported. Static field private static volatile java.lang.System$Logger jdk.internal.event.EventHelper.securityLogger is not properly marked for Unsafe access!                                  
[identity-1.0.0-SNAPSHOT-runner:134700]     (clinit):   2,120.21 ms,  6.77 GB                                                                                                                                                                                        
[identity-1.0.0-SNAPSHOT-runner:134700]   (typeflow): 107,127.73 ms,  6.77 GB                                                     
[identity-1.0.0-SNAPSHOT-runner:134700]    (objects):  70,590.48 ms,  6.77 GB                               
[identity-1.0.0-SNAPSHOT-runner:134700]   (features):   2,018.63 ms,  6.77 GB                                                
[identity-1.0.0-SNAPSHOT-runner:134700]     analysis: 194,886.87 ms,  6.77 GB                                                                                                                                                                                        
Error: Unsupported features in 3 methods                                                                                                                                                                                                                             
Detailed message:                                                                                                                                                                                                                                                    
Error: Field com.sun.xml.bind.v2.model.impl.TypeInfoImpl.upstream has declared type com.sun.xml.bind.v2.model.annotation.Locatable which is incompatible with types in state: 1TypeMObject<[0x00005C0F00005C10:!S:XmlSeeAlso$$ProxyImpl]>



Describe GraalVM and your environment:

    GraalVM version: 20.1.0 JDK major version: 11.0.7 OS: Debian Architecture: AMD64

**Comment by antonwiens** [comment]
Any news on this one?

**Comment by antonwiens** [comment]
I found the issue in my project:
I create a variable in a companion object in kotlin:
```
companion object {
        private val identityAssertionMarschaller: Marshaller by JAXBContext.newInstance(MyJaxbClass::class.java).createMarshaller()
}
```

This line is causing following error:

> Error: Field com.sun.xml.bind.v2.model.impl.TypeInfoImpl.upstream has declared type com.sun.xml.bind.v2.model.annotation.Locatable which is incompatible with types in state: 1TypeMObject<[0x000019E2000019E3:!S:XmlSeeAlso$$ProxyImpl]>

I don't know why this is caused. Does anyone know? @gsmet

**Comment by antonwiens** [comment]
I also found out, that the service file and impl class for JAXBContextFactory are not properly registered for JAXBContext.newInstance(...), i had to additionally register the when using quarkus-jaxb

**Comment by argenstijn** [comment]
Do you still have the same issue when using the laterst vrsion of graalvm,quarkus and quarkus-cxf?

**Comment by argenstijn** [comment]
Somehow weird:    Sometimes it work and sometimes it fails. 

Using 1.13.0.Final quarkus, graalv 21 and 

<dependency>
			<groupId>io.quarkiverse.cxf</groupId>
			<artifactId>quarkus-cxf</artifactId>
			<version>0.5</version>
		</dependency>

**Comment by argenstijn** [comment]
I include a simpleJaxb project which can reproduce this error constantly:
using: 
graalvm 21
quarkus 1.13.0
jaxb-extension

[simpleJaxb.zip](https://github.com/quarkusio/quarkus/files/6248830/simpleJaxb.zip)

**Comment by argenstijn** [comment]
This fixes  the problem but dunno what i am breaking then:

`
@TargetClass(className = "com.sun.xml.bind.v2.model.impl.TypeInfoImpl", onlyWith = Target_com_sun_xml_bind_v2_model_impl_TypeInfoImpl.Selector.class)
@Substitute
final class Target_com_sun_xml_bind_v2_model_impl_TypeInfoImpl {   
    
    
    static final class Selector implements BooleanSupplier {

        @Override
        public boolean getAsBoolean() {
            try {
                Class.forName("com.sun.xml.bind.v2.model.impl.TypeInfoImpl");
                return true;
            } catch (Exception e) {
                return false;
            }
        }
    }
}
`

**Comment by argenstijn** [comment]
Any update?

**Comment by argenstijn** [comment]
Seems to be fixed with Using 1.13.7.Final quarkus, graalv 21.0.X and
io.quarkiverse.cxf quarkus-cxf 0.8

Closing issue

**Comment by gsmet** [comment]
Thanks for the follow-up!
